app/src/main/java/org/wikipedia/activitytab/timeline/TimelineModule.kt

```diff
@@ -43,24 +44,43 @@ import org.wikipedia.util.DateUtil
 import org.wikipedia.views.imageservice.ImageService
 import java.util.Date
 
-// @TODO: MARK_ACTIVITY_TAB retrieve description and thumbnail for contributions through API
 @Composable
 fun TimelineModule(
-    modifier: Modifier = Modifier,
     timelineItem: TimelineItem,
     onItemClick: (TimelineItem) -> Unit
 ) {
-    Row(
+    Column(
         modifier = Modifier
-            .fillMaxWidth()
             .clickable(onClick = { onItemClick(timelineItem) })
-            .padding(horizontal = 16.dp, vertical = 8.dp),
+            .padding(horizontal = 16.dp, vertical = 8.dp)
+    ) {
+        TimelineHeader(
+            modifier = Modifier
+                .fillMaxWidth(),
+            timelineItem = timelineItem
+        )
+
+        if (timelineItem.activitySource == ActivitySource.EDIT) {
+            TimelineBottomView(
+                onViewChangesBtnClick = { onItemClick(timelineItem) }
+            )
+        }
+    }
+}
+
+@Composable
+fun TimelineHeader(
+    timelineItem: TimelineItem,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        modifier = modifier,
         horizontalArrangement = Arrangement.spacedBy(16.dp),
         verticalAlignment = Alignment.CenterVertically
     ) {
         val icon = when (timelineItem.activitySource) {
             ActivitySource.EDIT -> R.drawable.ic_mode_edit_white_24dp
-            ActivitySource.SEARCH -> R.drawable.search_bold
+            ActivitySource.SEARCH -> R.drawable.outline_search_24
```

Comment:
I believe you can use the existing `ic_search_white` instead of importing a new one from IDE.